libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -472,6 +472,18 @@ bool AP_Relay::arming_checks(size_t buflen, char *buffer) const
             }
             return false;
         }
+
+        // Each pin can only be used by a single relay
+        for (uint8_t j=i+1; j<ARRAY_SIZE(_params); j++) {
+            if (!function_valid((AP_Relay_Params::FUNCTION)_params[j].function.get())) {
```

Comment:
I guess the enabled() method could have been used here as well.

Reply:
Enabled by instance only returns true if its using the "relay" function. Its all a bit odd because we still support numbered relays for RC and MAVLink but internally we use the new relay functions.